Add '--address' flag to 'linkerd dashboard'.

bmcstdio opened the pull request:
This PR adds a `--address` flag to `linkerd dashboard`, with default value `localhost`, as requested in #3100.

Closes #3100.

grampelberg commented:
Awesome! I ran into this last week with my chromebook and ended up doing something way more crazy.

ihcsim commented:
@bmcstdio Would you mind rebasing your branch with `master`, and run the [integration test](https://github.com/linkerd/linkerd2/blob/master/TEST.md#integration-tests)? On my machine, I need to make the following changes to get the tests to pass. (Might be related to the change to use `portfoward.NewOnAddress()`?)

```diff
diff --git a/test/install_test.go b/test/install_test.go
index 66df67b3..5bbbf109 100644
--- a/test/install_test.go
+++ b/test/install_test.go
@@ -335,7 +335,7 @@ func TestInstallSP(t *testing.T) {
 
 func TestDashboard(t *testing.T) {
 	dashboardPort := 52237
-	dashboardURL := fmt.Sprintf("http://127.0.0.1:%d", dashboardPort)
+	dashboardURL := fmt.Sprintf("http://localhost:%d", dashboardPort)
 
 	outputStream, err := TestHelper.LinkerdRunStream("dashboard", "-p",
 		strconv.Itoa(dashboardPort), "--show", "url")
```
Let me know if you have any questions about the integration test. Thanks!

bmcstdio commented:
@ihcsim @siggy thank you for taking a look! I've rebased with `master`, fixed that issue in `install_test.go` and ran the integration tests:

```
bin/test-run `pwd`/bin/linkerd
Checking the linkerd binary...[ok]
Checking if there is a Kubernetes cluster available...[ok]
==================RUNNING ALL TESTS==================
Testing Linkerd version [git-c8135979] namespace [l5d-integration] k8s-context []
Test script: [install_test.go] Params: [--upgrade-from-version=stable-2.5.0 --linkerd-namespace=l5d-integration-upgrade]
ok      command-line-arguments  137.085s
namespace/l5d-integration-tiller created
namespace/l5d-integration-tiller labeled
clusterrolebinding.rbac.authorization.k8s.io/l5d-integration-tiller:tiller-cluster-admin created
clusterrolebinding.rbac.authorization.k8s.io/l5d-integration-tiller:tiller-cluster-admin labeled
$HELM_HOME has been configured at /Users/bmcstdio/.helm.

Tiller (the Helm server-side component) has been installed into your Kubernetes Cluster.

Please note: by default, Tiller is deployed with an insecure 'allow unauthenticated users' policy.
To prevent this, run `helm init` with the --tiller-tls-verify flag.
For more information on securing your installation see: https://docs.helm.sh/using_helm/#securing-your-helm-installation
Hang tight while we grab the latest from your chart repositories...
...Unable to get an update from the "local" chart repository (http://127.0.0.1:8879/charts):
        Get http://127.0.0.1:8879/charts/index.yaml: dial tcp 127.0.0.1:8879: connect: connection refused
...Successfully got an update from the "cloudflare" chart repository
...Successfully got an update from the "harbor" chart repository
...Successfully got an update from the "openfaas" chart repository
...Successfully got an update from the "elastic" chart repository
...Successfully got an update from the "stable" chart repository
...Successfully got an update from the "incubator" chart repository
Update Complete.
Saving 1 charts
Deleting outdated charts
Test script: [install_test.go] Params: [--linkerd-namespace=l5d-integration-helm --helm-path=/Users/bmcstdio/Stuff/linkerd2/bin/helm --helm-chart=/Users/bmcstdio/Stuff/linkerd2/charts/linkerd2 --helm-release=l5d-integration-test --tiller-ns=l5d-integ
ration-tiller]
ok      command-line-arguments  116.398s
release "l5d-integration-test" deleted
namespace/l5d-integration-helm condition met
Tiller (the Helm server-side component) has been uninstalled from your Kubernetes Cluster.
clusterrolebinding.rbac.authorization.k8s.io "l5d-integration-tiller:tiller-cluster-admin" deleted
l5d-integration-tiller
namespace "l5d-integration-tiller" deleted
Test script: [install_test.go] Params: [--linkerd-namespace=l5d-integration]
ok      command-line-arguments  129.715s
Test script: [egress
get
inject
routes
serviceprofiles
stat
tap] Params: [--linkerd-namespace=l5d-integration]
ok      github.com/linkerd/linkerd2/test/egress 52.366s
ok      github.com/linkerd/linkerd2/test/get    45.013s
ok      github.com/linkerd/linkerd2/test/inject 56.341s
ok      github.com/linkerd/linkerd2/test/routes 4.468s
ok      github.com/linkerd/linkerd2/test/serviceprofiles        76.272s
ok      github.com/linkerd/linkerd2/test/stat   12.436s
ok      github.com/linkerd/linkerd2/test/tap    63.640s

=== PASS: all tests passed
```

grampelberg commented:
@bmcstdio we'd love to send you some contributor swag! Mind filling out the [form](https://docs.google.com/forms/d/e/1FAIpQLSdXA7kAl6oyUQ4DFRjiDc2biWlvkhNirrMwsXaQ8V3Uh8Yj5A/viewform?usp=sf_link)?

bmcstdio commented:
@grampelberg done, thank you so much! 🙂
